### PR TITLE
oop-creating_objects_from_classes-hard-q1-modifications

### PR DESCRIPTION
**Pull request by Hedgemon4**
## PR Checklist
- [ ]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [ ]  Are the assertion fail messages good?
    - [ ]  Logical (does it make sense?)
    - [ ]  Readable (can a student understand what’s wrong?)
    - [ ]  Not too feed-y (does it give away the answer?)
    - [ ]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [ ]  Specific (does the fail message relate to the question?)
- [ ]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [ ]  EOF newlines
- [ ]  Do the comments for Parsons follow the correct convention?
- [ ]  Is the question being tested sufficiently?
    - [ ]  Are there representative cases to test the program?
    - [ ]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [ ]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

File: src/oop/creating_objects_from_classes/hard/q1/MainTest.java [hunk 1]
```diff
@@ -1,59 +1,134 @@
...
+
+class Dog {
+    private String name;
+    private int age;
+    private double weight;
+    private static Dog dog;
+    private static boolean constructorCalled = false;
+    private static String constructorName;
+    private static double constructorWeight;
+    private static int constructorAge;
```
Comment (te-sa) [inline src/oop/creating_objects_from_classes/hard/q1/MainTest.java:80]: Make those final and leave comment about purpose

**Review by te-sa — CHANGES_REQUESTED** [review]

File: src/oop/creating_objects_from_classes/hard/q1/MainTest.java (hunk 1 above)
Comment (Hedgemon4) [reply src/oop/creating_objects_from_classes/hard/q1/MainTest.java:80]: You can't do this because these values get changed in the course of the test, so they can't be final
